packages/vulcan-core/lib/modules/containers/withSingle.js

```diff
@@ -76,3 +82,15 @@ export default function withSingle(options) {
     }
   });
 }
+
+export function useSingle(options, documentId, extraVariables) {
```

Comment:
Maybe in the next PR rather than this one, but, the options list should be explicit. (And `documentId` and `extraVariables` probably shouldn't be separated out.)

Reply:
Yeah, this API is probably the one we want to think about the most. 

Given that we pass the options list through to some magic functions I don't think I can make it explicit, or at least would have to chase a lot of references to figure out all the valid arguments. I guess I can put in the work to do that, but not even sure whether it makes the thing easier to read, because of the other functions the options argument gets passed to. 

Combining documentId and extravariables into an object seems fine to me, though the most frequent use-case will obviously be to just provide a documentId without extraVariables, and this would make that use-case harder.

Reply:
I don't think getting the full options list is that hard--it gets passed, but the dataflow doesn't go very far. It definitely does make it easier to read, from the perspective of someone figuring out how to call this; currently there is no list anywhere of what arguments it takes, and that's very important to have.

Reply:
I don't see how it makes the "provide a documentId and no extraVariables" use case any harder? It's just replacing positional parameters with named ones.